### PR TITLE
Add ADR for how we structure future ones

### PR DESCRIPTION
**Pull request by benthorner**
https://trello.com/c/Xd47oJKZ/535-remove-the-deliveryattempt-model

### CODE REVIEW

File: docs/adr/adr-005-record-architecture-decisions.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# 1. Record architecture decisions
```
Comment (kevindew) [inline docs/adr/adr-005-record-architecture-decisions.md:1]: This should be 5

File: docs/adr/adr-005-record-architecture-decisions.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+Date: 2020-10-21
+
+## Context
+
+While we have already recorded architectural decisions for this project, the structure of these documents has been arbitrary, since there was no guidance to follow. Recently we found that creating a new ADR was [hindered](https://github.com/alphagov/email-alert-api/pull/1441#discussion_r508729384) by wanting to be consistent with the structure of previous ADRs.
```
Comment (kevindew) [inline docs/adr/adr-005-record-architecture-decisions.md:7]: The structure wasn't arbitrary it was based on: https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_for_alexandrian_pattern.md

File: docs/adr/adr-005-record-architecture-decisions.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+We will adopt a structure for future ADRs that matches this document - the exemplar in [our Rails app conventions](https://docs.publishing.service.gov.uk/manual/conventions-for-rails-applications.html#documenting-your-decisions). We will not change how the files are named, to avoid breaking links that were never expected to change.
+
+## Status
+
+Accepted
+
+## Consequences
+
+None.
```
Comment (kevindew) [inline docs/adr/adr-005-record-architecture-decisions.md:21]: There's a consequence that the first 4 ADRs are in a different format.

**Review by kevindew — COMMENTED** [review]
Thanks for raising just a few minor comments.

File: docs/adr/adr-005-record-architecture-decisions.md (hunk 1 above)
Comment (benthorner) [reply docs/adr/adr-005-record-architecture-decisions.md]: Oops - fixed.

File: docs/adr/adr-005-record-architecture-decisions.md (hunk 2 above)
Comment (benthorner) [reply docs/adr/adr-005-record-architecture-decisions.md]: Oh fair enough. It was an attempt to say there was no documented reasoning behind the layout - at least, not that I could find. Anyway, it was a minor point - removed.

File: docs/adr/adr-005-record-architecture-decisions.md (hunk 3 above)
Comment (benthorner) [reply docs/adr/adr-005-record-architecture-decisions.md]: Oh yes, good point - I usually read "consequences" as "actions". I've added that in.

**Comment by benthorner** [comment]
@kevindew thanks for looking through. I hadn't previously realised that you were the author of all of the previous ADRs. So to have me come along and suggest a somewhat breaking change must be a bit frustrating. Thanks for your patience so far. I'm open to us sticking with the existing structure if you'd prefer.

File: docs/adr/adr-005-record-architecture-decisions.md (hunk 2 above)
Comment (kevindew) [reply docs/adr/adr-005-record-architecture-decisions.md]: Yeah not sure why the reason for using that format wasn't documented - that's my bad. I have a recollection of, at the time, it being less common to have ADRs across the programme so it felt more free form and there was some contention around "status" given it's pretty much a given they're accepted to be merged.

**Review by kevindew — APPROVED** [review]
Thanks @benthorner and appreciate the concern. I hadn't realised I was the lone ADR author for this repo. I don't have an attachment to the past ones structure, from todays perspective it feels inconsistent.

**Pull request merged**